tools/IceRpc.ProtocGen/ClientGenerator.cs

```diff
@@ -107,15 +107,15 @@ internal static string GenerateImplementation(ServiceDescriptor service)
 [global::System.Obsolete]";
         }
         clientImplementation += @$"
-public readonly partial record struct {clientImplementationName} : I{service.Name.ToPascalCase()}
+public readonly partial record struct {clientImplementationName} : I{service.Name.ToPascalCase()}, IProtobufClient
 {{
     /// <summary>Represents the default path for IceRPC services that implement Protobuf service
     /// <c>{service.FullName}</c>.</summary>
     public const string DefaultServicePath = ""/{service.FullName}"";
 
     /// <summary>Gets or initializes the encode options, used to customize the encoding of payloads created from this
     /// client.</summary>
-    ProtobufEncodeOptions? EncodeOptions {{ get; init; }}
+    public ProtobufEncodeOptions? EncodeOptions {{ get; init; }} = null;
```

Comment:
It's not clear to me why we need the `= null` since we don't have an explicit parameterless constructor.

Reply:
You are right it is not required.